internal/resources/aws/elasticache_cluster.go

```diff
@@ -104,3 +121,56 @@ func (r *ElastiCacheCluster) backupStorageCostComponent() *schema.CostComponent
 		},
 	}
 }
+
+// ElastiCache implementation of reservationResolver
+type elasticacheReservationResolver struct {
+	term          string
+	paymentOption string
+	cacheNodeType string
+}
+
+func newElasticacheReservationResolver(term, paymentOption, cacheNodeType string) reservationResolver {
+	return &elasticacheReservationResolver{
+		term:          term,
+		paymentOption: paymentOption,
+		cacheNodeType: cacheNodeType,
+	}
+}
+
+func (r elasticacheReservationResolver) IsElasticacheReservedNodeLegacyOffering() bool {
```

Comment:
```suggestion
func (r elasticacheReservationResolver) isElasticacheReservedNodeLegacyOffering() bool {
```

as it's not exported outside this package

Reply:
Updated in the last commit 👍